boards.txt

```diff
@@ -25,26 +25,26 @@ generic.serial.disableRTS=true
 
 generic.build.mcu=esp8266
 generic.build.f_cpu=80000000L
-generic.build.f_crystal=1
 generic.build.board=ESP8266_ESP01
 generic.build.core=esp8266
 generic.build.variant=generic
 generic.build.flash_mode=qio
 generic.build.spiffs_pagesize=256
 generic.build.debug_port=
 generic.build.debug_level=
+generic.build.extra_flags=-DF_CRYSTAL={build.f_crystal}
```

Comment:
Arduino IDE expands it before compiling, into console I get the correct value: `-DF_CRYSTAL=24000000`

```shell
Compiling sketch...
"/Users/sergio/Library/Arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/Users/sergio/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/tools/sdk/include" "-I/Users/sergio/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/tools/sdk/lwip/include" "-I/Users/sergio/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/tools/sdk/libc/xtensa-lx106-elf/include" "-I/var/folders/k4/q1bl6yfs2rv0n006w3nh0rk40000gn/T/arduino_build_308216/core" -c -w -Os -g -mlongcalls -mtext-section-literals -fno-exceptions -fno-rtti -falign-functions=4 -std=c++11 -MMD -ffunction-sections -fdata-sections -DF_CPU=80000000L -DLWIP_OPEN_SRC   -DARDUINO=10801 -DARDUINO_ESP8266_ESP01 -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="ESP8266_ESP01"  -DF_CRYSTAL=24000000 "-I/Users/sergio/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/cores/esp8266" "-I/Users/sergio/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/variants/generic" "/var/folders/k4/q1bl6yfs2rv0n006w3nh0rk40000gn/T/arduino_build_308216/sketch/sketch_apr11a.ino.cpp" -o "/var/folders/k4/q1bl6yfs2rv0n006w3nh0rk40000gn/T/arduino_build_308216/sketch/sketch_apr11a.ino.cpp.o"
```

This kind of expression is needed if you want to keep the *Crystal Freq.* Menu to allow users to select crystal frequency by themselves. 

For the same reason I should also change this:

`arduino-esp8266.build.extra_flags=-DF_CRYSTAL={build.f_crystal}`

Let me know how to proceed and move forward ;)